src/core/server/saved_objects/routes/integration_tests/import.test.ts

```diff
@@ -187,7 +187,7 @@ describe('POST /internal/saved_objects/_import', () => {
           references: [],
           error: {
             statusCode: 409,
-            message: 'version conflict, document already exists',
+            message: 'Saved object [index-pattern/my-pattern] conflict',
```

Comment:
Updated in https://github.com/elastic/kibana/pull/54605/commits/e2f5d6573c45086c5dc1bbca045228590e35fa0b.

Reply:
What was the reasoning behind changing the error message? Is that caused by a change in the ES apis we are using, or is this a message from our codebase?

Reply:
It's a message from our codebase, I changed it to be consistent with other Conflict errors from other operations.

Reply:
Conflicts can also occur if you're overwriting an existing object and setting a namespace that's different from the current one (or another way to put it: overwriting an existing object in a different namespace). This change makes these two errors the same.

Reply:
Before, some methods would return a 409 without a message and some would return this "version conflict, document already exists" message. I think this message is misleading because `version_conflict_engine_exception` could be an id conflict or a version conflict.

Maybe we should make the "exists in a different namespace" conflict more clear as it can be confusing to get a conflict when you're specifying `overwrite=true`.

Reply:
> Maybe we should make the "exists in a different namespace" conflict more clear as it can be confusing to get a conflict when you're specifying `overwrite=true`.

The only way that I think this situation will happen organically is if a multi-namespace saved object exists in one space, and the user exports the document and tries to import it into another space.

In #62553 we will need to differentiate between these conflicts on the server side. I planned to add some metadata to the error so we can identify these types of "unresolvable conflicts". We *could* make it an actual different 409 error message instead, but I thought that would involve changing more logic in consumers (now they would all need to check for two different types of conflicts).

Reply:
> The only way that I think this situation will happen organically is if a multi-namespace saved object exists in one space, and the user exports the document and tries to import it into another space.

We will probably have to surface this error to end-users, but I was thinking mostly of differentiating the errors as a form of developer documentation. If you're developing or writing tests you could easily run into this error and not understand why overwrite isn't working. Then it might be really useful to know whether it was a version conflict, id conflict or "exists in another namespace" conflict.

We don't want code to rely on the message itself, it's just a hint for developers, so I like the idea of using metadata and agree that consuming code will probably only ever need to know "can I retry with overwrite=true".

If you're anyway revisiting these errors in https://github.com/elastic/kibana/pull/62553 I'm 👍 for making the error message more specific as part of that.